src/mono/mono/component/debugger-agent.c

```diff
@@ -6945,6 +6946,87 @@ valid_memory_address (gpointer addr, gint size)
 	return ret;
 }
 
+static MonoAssembly* 
+find_assembly_by_name (char* assembly_name)
+{
+	//we get 'foo.dll' but mono_assembly_load expects 'foo' so we strip the last dot
+	char *lookup_name = g_strdup (assembly_name);
+	for (int i = ((int)strlen (lookup_name) - 1); i >= 0; --i) {
```

Comment:
Would strrchr work here? Or maybe there in some existing code in mono/eglib to get just the filename?

Reply:
I didn't change this code, I just moved it.